src/promptflow/promptflow/batch/_batch_engine.py

```diff
@@ -253,6 +256,9 @@ def _copy_previous_run_result(
                 previous_run_info = resume_from_run_storage.load_flow_run_info(i)
 
                 if previous_run_info and previous_run_info.status == Status.Completed:
+                    # Change the root_run_id and parent_run_id to the new run_id
```

Comment:
Could you share more context how UI depends on these two IDs? We need to understand the run info interactions more clearly.